torchdynamo/eval_frame.py

```diff
@@ -524,6 +555,66 @@ def output(self, target, args, kwargs):
         graph,
     ).transform()
 
+    if guard_args:
+        id_to_arg_mapping = dict()
+        for i in range(len(args)):
+            a = args[i]
+            id_to_arg_mapping[id(a)] = (i, a)
+        placeholders = [n for n in new_graph.graph.nodes if n.op == "placeholder"]
+        if len(placeholders) > 0:
+            for guard in out_guards:
```

Comment:
Why not just guard the shapes of all inputs (one for each placeholder)?  When would this matching logic result in a different outcome?

Reply:
Not all inputs are tensors, and this is actually a bit of a setup for symbolic shapes, in that we use the guard. Right now we just use it for id - but in the future, I was hoping to have the guard be the source of truth for the sym shape info.